BoardStateClassTests.py

```diff
@@ -0,0 +1,160 @@
+import unittest
+from ColumnClass import Column
+from PlayerClass import Player
+from BoardStateClass import Board_State
+
+
+class Test_Board_State(unittest.TestCase):
+
+    def setUp(self):
+        self._board_width = 5
+        self._board_height = 10
+        self._win_number = 4
+        self.player1 = Player("Jarvis", "yellow")
+        self.player2 = Player("Ultron", "red")
+
```

Comment:
consider using the given when then naming convention for the tests. This leads to more behaviour focussed tests rather than implementation. question, why are you testing create_columns as this is a "private" method? all you should care about is how the public methods behave.